os/arch/arm/src/armv8-m/up_blocktask.c

```diff
@@ -144,7 +144,7 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 	if (switch_needed) {
 #ifdef CONFIG_ARMV8M_TRUSTZONE
 		if (tcb->tz_context) {
-			TZ_StoreContext_S(tcb->tz_context);
+			TZ_StoreContext_S(rtcb->tz_context);
```

Comment:
Context switch will be needed only if the task that is being blocked is at the head of the ready-to-run list  (otherwise the head won't change and no context switch is needed). So, both rtcb and tcb will be same in this case.